### ISSUE TITLE
Basic test produces error `invalid section ID: 32`

### ISSUE THREAD

**Issue by ramiro050** [openxla/openxla-pjrt-plugin#233]
## Expected Behavior

Running `JAX_PLATFORMS=iree_cpu python test/test_simple.py` should not result in any errors.

## Actual Behavior

Running the script results in the following output:

```
Platform 'iree_cpu' is experimental and not all JAX functionality may be correctly supported!
[IREE-PJRT] DEBUG: Using IREE compiler binary: /usr/local/google/home/ramiroleal/openxla/iree-build/lib/libIREECompiler.so
[IREE-PJRT] DEBUG: Compiler Version: <unknown> (API version 1.2)
[IREE-PJRT] DEBUG: Partitioner was not enabled. The partitioner can be enabled by setting the 'PARTITIONER_LIB_PATH' config var ('IREE_PJRT_PARTITIONER_LIB_PATH' env var)
[IREE-PJRT] DEBUG: CPU driver created
loc("-":0:0): error: invalid section ID: 32
loc("-":0:0): error: invalid section ID: 32
[ 4951  9902 14853 19804 24755 29706 34657 39608 44559]
[ 49510  99020 148530 198040 247550 297060 346570 396080 445590]
```

After the [most recent update ](https://github.com/openxla/openxla-pjrt-plugin/commit/f7538e2dd9134f24019f4654092b448da0944c8f)to nightly dependencies, the `loc("-":0:0): error: invalid section ID: 32` started appearing. See [CI output](https://github.com/openxla/openxla-pjrt-plugin/actions/runs/5591694199/jobs/10223124167#step:9:17).

## Specifications

- Version: f7538e2dd9134f24019f4654092b448da0944c8f

**Comment by jpienaar** [comment]
I recall seeing this recently and not sure if due to my change (I don't think it should be ...). I believe the bytecode version is pinned, but it's also weird that it passes while returning the error.

**Comment by ramiro050** [comment]
This is the only related information I was able to find: https://github.com/llvm/llvm-project/issues/62932

**Comment by jpienaar** [comment]
[error.mlirbc.zip](https://github.com/openxla/openxla-pjrt-plugin/files/12329948/error.mlirbc.zip)
 is an example from 

```
JAX_PLATFORMS=iree_cpu  python -c "import jax; a = jax.numpy.asarray([1, 2, 3, 4, 5, 6, 7, 8, 9]); print(a + a);"
```

**Comment by jpienaar** [comment]
Seems this is due to padding with spaces and size not being correctly set.
